core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/SchemaElementDefinition.java

```diff
@@ -704,9 +704,18 @@ public CHILD_CLASS merge(final ELEMENT_DEF elementDef) {
             elDef.ingestAggregatorCache = null;
             elDef.queryAggregatorCacheMap.clear();
 
-            elDef.groupBy = new LinkedHashSet<>(elementDef.groupBy);
-            elDef.parents = null != elementDef.parents ? new LinkedHashSet<>(elementDef.parents) : null;
-            elDef.description = elementDef.description;
+            if (!elementDef.groupBy.isEmpty()) {
+                elDef.groupBy = new LinkedHashSet<>(elementDef.groupBy);
```

Comment:
Can elementDef.groupBy be null? if so you will get a NPE thrown.  Might be worth adding the null != elementDef.groupBy check?